File: backend-side/staff-service/src/main/java/com/sid/staff_service/entity/Employee.java
package com.sid.staff_service.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;


@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Employee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String firstname;
    private String lastname;
    private String email;
    private Date birthDate;
    private String phonenumber;
    private String adress;
    private Double salary;

    @ManyToOne
    private Team team;
    @ManyToOne

    private Departement departement;

    @Enumerated(EnumType.STRING)
    private Level level;
}

File: frontend-side/src/app/core/models/Employee.ts
import { Contract } from "./Contract";
import { Department } from "./department";
import { Team } from "./Team";

export class Employee {
  id?: number;
  firstname: string;
  lastname: string;
  email: string;
  phoneNumber: string;
  salary: number;
  level: string;
  adress: string;
  department: Department;
  team: Team;
}
